src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemorySegment.java

```diff
@@ -331,22 +331,22 @@ default MemorySegment asSlice(long offset) {
      * this case, or when no overlap occurs, {@code null} is returned.
      *
      * @param other the segment to test for an overlap with this segment.
-     * @return a new slice of this segment, or {@code null} if no overlap occurs.
+     * @return a slice of this segment, or {@code null} if no overlap occurs.
      */
     MemorySegment asOverlappingSlice(MemorySegment other);
 
     /**
-     * Returns the offset, in bytes, from this segment to the other segment.
+     * Returns the offset, in bytes, of the provided segment, relative to this
+     * segment.
      *
-     * <p>The offset is relative to the {@linkplain #address() base address} of
-     * this segment and can be a negative or positive value. If the segments
-     * share the same base address, {@code 0} is returned.
+     * <p>The offset is relative to the base address of this segment and can be
```

Comment:
Perhaps we could add something like:
```For instance, if both segments are native segments, the resulting offset can be computed as follows:
other.baseAddress().toRawLongValue() - segment.baseAddress().toRawLongValue()
```
So that it hints at how the method fully operates. Then you can conclude with the sentence which says we return `0` if base is the same. I'd also add another sentence to say that if `other` is a slice of this segment, then the offset is always `0 <= x <= byteSize()`.